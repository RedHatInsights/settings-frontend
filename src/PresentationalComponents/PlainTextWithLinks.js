import React from 'react';
import PropTypes from 'prop-types';
import { Text, TextVariants } from '@patternfly/react-core';

export const PlainTextWithLinks = ({ text, linkProps }) => {
  let linkIx = 0;
  const items = text.split(/<link>/).reduce((acc, curr) => {
    let slice = curr.split(/<\/link>/);
    if (slice.length === 2) {
      slice[0] = (
        <Text
          key={`link-${linkIx}`}
          rel="noreferrer noopener"
          className="pf-u-ml-md"
          component={TextVariants.a}
          target="_blank"
          {...linkProps[linkIx]}
        >
          {slice[0]}
        </Text>
      );
      linkIx++;
    }

    return [...acc, ...slice];
  }, []);
  return <Text component={TextVariants.p}>{items}</Text>;
};

PlainTextWithLinks.propTypes = {
  text: PropTypes.string.isRequired,
  linkProps: PropTypes.arrayOf(PropTypes.object).isRequired,
};
