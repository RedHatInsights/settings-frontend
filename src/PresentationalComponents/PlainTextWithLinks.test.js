import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PlainTextWithLinks } from './PlainTextWithLinks';

it('plain text hack - render', () => {
  const wrapper = shallow(
    <PlainTextWithLinks
      text="hello world <link>Read here</link> and good bye <link>Learn more</link>"
      linkProps={[
        { href: 'http://github.com' },
        { href: '#', className: 'ccc' },
      ]}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
