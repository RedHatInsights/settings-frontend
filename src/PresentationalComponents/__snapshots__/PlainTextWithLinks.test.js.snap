// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plain text hack - render 1`] = `
<Text
  component="p"
>
  hello world 
  <Text
    className="pf-u-ml-md"
    component="a"
    href="http://github.com"
    key="link-0"
    rel="noreferrer noopener"
    target="_blank"
  >
    Read here
  </Text>
   and good bye 
  <Text
    className="ccc"
    component="a"
    href="#"
    key="link-1"
    rel="noreferrer noopener"
    target="_blank"
  >
    Learn more
  </Text>
</Text>
`;
